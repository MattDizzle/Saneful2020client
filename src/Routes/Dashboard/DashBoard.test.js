import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import DashBoard from './index';


describe(`DashBoard Component`, () => {
    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            const div = document.createElement('div');
            ReactDOM.render(
                <BrowserRouter>
                    <DashBoard />
                </BrowserRouter>
                , div);
            ReactDOM.unmountComponentAtNode(div);
        });
    });
    
    describe(`Snapshot test`, () => {
        it(`Renders the UI as expected`, () => {
            const tree = renderer
                .create(
                    <BrowserRouter>
                        <DashBoard />
                    </BrowserRouter>
                )
                .toJSON()
            expect(tree).toMatchSnapshot()
        });
    });
})