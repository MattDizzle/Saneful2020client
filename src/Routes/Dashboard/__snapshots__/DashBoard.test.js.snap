// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashBoard Component Snapshot test Renders the UI as expected 1`] = `
<section
  className="Dashboard"
>
  <h2>
    Dashboard
  </h2>
  <p>
    Welcome back 
    !
  </p>
  <h3>
    Loading...
  </h3>
  <div
    className="write-up"
  >
    <h3>
      The Game
    </h3>
    <p>
      You are living in Los Pixels in the year 2020 during a global pandemic. Your goal is to survive in self quarantine. 
    </p>
    <h3>
      How to Play
    </h3>
    <p>
      There are 3 metrics that you must maintain: Health, Sanity, Money - all three deplete with time.
    </p>
    <p>
      If health or sanity drop below 0 you will perishâ€¦
    </p>
    <p>
      There are multiple ways to increase and decrease your health, sanity and money and take a few lives to find a good survival strategy... but you are probably smarter than you look!
    </p>
    <p>
       Beware, difficulty increases with each passing day. 
    </p>
  </div>
</section>
`;
